src/journal/journald-file.h

```diff
@@ -0,0 +1,41 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+#pragma once
+
+#include "journal-file.h"
+
+typedef struct {
+        JournalFile *file;
```

Comment:
I don't understand why this is a pointer. Why not just include the JournalFile structure directly here?

Reply:
Good point, no reason at all actually, let me change this

Reply:
Actually, I'm slightly confused on how to deal with this. `journal_file_open()` takes a return argument of `JournalFile **` and allocates the `JournalFile` instance itself on the heap. I'm assuming this happens because `JournalFile` is a somewhat large struct and we want to avoid copying it all the time. However, in `JournaldFile`, we want to embed it instead of doing double indirection. But if we embed it, we can't use `journal_file_open` to initialize the instance anymore because we'd be doing an unnecessary allocation and then copying it into `JournaldFile`.

Not sure how we usually deal with these kind of situations. Do we have any examples in the codebase where we embed a large struct that's usually allocated on the heap inside another struct where we don't want extra indirection?

Reply:
Found a precedent in `struct Home` in homed-home.h which embeds an `UserRecord` instance but also does so with pointer indirection.